test/framework/src/main/java/org/elasticsearch/test/ClusterServiceUtils.java

```diff
@@ -53,12 +53,12 @@ public static ClusterService createClusterService(ThreadPool threadPool, Discove
                 threadPool, () -> localNode);
         clusterService.setNodeConnectionsService(new NodeConnectionsService(Settings.EMPTY, null, null) {
             @Override
-            public void connectToNodes(Iterable<DiscoveryNode> discoveryNodes) {
+            public void connectToNodes(DiscoveryNodes discoveryNodes) {
                 // skip
             }
 
             @Override
-            public void disconnectFromNodesExcept(Iterable<DiscoveryNode> nodesToKeep) {
+            public void disconnectFromNodesExcept(DiscoveryNodes nodesToKeep) {
```

Comment:
> why do we change these? Isn't Iterable more generic and would still accept DiscoveryNodes?

I can change it back, but I did it to be consistent with connectToNodes which needs the total size of the collection. Since this is only used with DiscoNodes, I felt consistency is clearer. As said - I'll roll it back if you prefer.

> disconnectFromNodesExcept sounds like it would only be executed on a subset of and actual DiscoveryNodes instance

Naming are hard. What it does is disconnecting from all nodes that are not part of the supplied disconodes parameter. We pass the disco nodes of the current cluster state directly, without any modification.